src/service/api.tsx

```diff
@@ -134,7 +138,51 @@ const api = {
     return obj.data.results;
   },
 
-  changeImgLinks(url: string, size: string) {
+  async getSearchFileredList(dataFilter:
+    {
+      from: number,
+      to: number,
+      genre: string | null,
+      page: number
+    } = {
+      from: 1900,
+      to: 2020,
+      genre: null,
+      page: 1,
+    }
+  ) {
+
+    function setFilteredData() {
+      const { from, to } = dataFilter
+      if (from < to) {
+        return `&primary_release_date.gte=${from}-01-01&primary_release_date.lte=${to}-01-01`
+      } else if (from > to) {
+        return `&primary_release_date.gte=${to}-01-01&primary_release_date.lte=${from}-01-01`
+      }
+      return `&primary_release_date.gte=${from}-01-01`
+    }
+
+    function setGenre() {
+      const { genre } = dataFilter;
+      if (genre !== null) {
```

Comment:
!genre will be enough

Reply:
On the comment above, that is fool-protection, and it works good :)